### ISSUE TITLE
ActivityBase.ActivityTask not initialized for local activities

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#786]
The `ActivityBase.ActivityTask` property is being set when activities are started but the fields are not being initialized.  There's a TODO in the source code for this.

We're going to need to add additional properties to `ActivityInvokeRequest` so that the **cadence-proxy** can pass this information to the .NET side.

**Comment by johncburns1** [comment]
I'll look into this on **cadence-proxy** and push those changes to `cadence` branch

**Comment by jefflill** [comment]
Cool.  I just merged some changes from **master** into **cadence.**

We're looking to be able to set the properties in the class below so the user's activity code has some context about where/how it's being executed:
```
    /// <summary>
    /// Holds information about an executing activity.
    /// </summary>
    public class ActivityTask
    {
        /// <summary>
        /// The opaque base-64 encoded activity task token.
        /// </summary>
        public string TaskToken { get; internal set; }

        /// <summary>
        /// The parent workflow type name.
        /// </summary>
        public string WorkflowTypeName { get; internal set; }

        /// <summary>
        /// The parent workflow domain.
        /// </summary>
        public string WorkflowDomain { get; internal set; }

        /// <summary>
        /// The parent workflow execution details.
        /// </summary>
        public WorkflowExecution WorkflowExecution { get; internal set; }

        /// <summary>
        /// The activity ID.
        /// </summary>
        public string ActivityId { get; internal set; }

        /// <summary>
        /// The activity type name.
        /// </summary>
        public string ActivityTypeName { get; internal set; }

        /// <summary>
        /// The activity task list.
        /// </summary>
        public string TaskList { get; internal set; }

        /// <summary>
        /// The maximum time between heartbeats.  0 means no heartbeat needed.
        /// </summary>
        public TimeSpan HeartbeatTimeout { get; internal set; }

        /// <summary>
        /// Time (UTC) when the activity was scheduled.
        /// </summary>
        public DateTime ScheduledTimeUtc { get; internal set; }

        /// <summary>
        /// Time (UTC) when the activity was started.
        /// </summary>
        public DateTime StartedTimeUtc { get; internal set; }

        /// <summary>
        /// Time (UTC) when the activity will timeout.
        /// </summary>
        public DateTime DeadlineTimeUtc { get; internal set; }

        /// <summary>
        /// Indicates how many times the activity was been restarted.  This will be zero
        /// for the first execution, 1 for the second, and so on.
        /// </summary>
        public int Attempt { get; internal set; }
    }
```

**Comment by jefflill** [comment]
It looks like we actually implemented this via `ActivityGetInfoRequest/ActivityGetInfoReply` messages and it appears to be working for regular activities.  This is not implemented for local activities and believe that's the correct behavior.

I'm going to modify `ActivityBase.ActivityTask` so it throws an exception when called on a local activity.

**Comment by jefflill** [comment]
Fixed for v2.1.0

**Comment by jefflill** [comment]
After thinking about this some more, my last comment was incorrect.  `ActivityTask` should be initialized for local activities as well.

**Comment by jefflill** [comment]
This required a bit of internal refactoring.  The `ActivityBase` class defined `ContextId` as a `long?` where local activities will have this set this to NULL and `IsLocal` just checked for NULL.

`ContextId==NULL` was preventing the **cadence-proxy** query from working because we need the context ID for that.  I refactored the `ContextId` to be a non-nullable `long` and explicitly set the `IsLocal` property now.  Now I can obtain the activity task information for both normal and local activities.

**Comment by jefflill** [comment]
**FIXED** for v2.1.0
